### PR TITLE
[FIX](map-type)fix map regress test & create mapTypeInfo without delete

### PR DESCRIPTION
**Pull request by amorynan**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/olap/types.cpp [hunk 1]
```diff
@@ -259,7 +259,7 @@ TypeInfoPtr get_type_info(const TabletColumn* col) {
         const auto* val_column = &col->get_sub_column(1);
         TypeInfoPtr value_type = get_type_info(val_column);
         MapTypeInfo* map_type_info = new MapTypeInfo(std::move(key_type), std::move(value_type));
-        return create_static_type_info_ptr(map_type_info);
+        return create_dynamic_type_info_ptr(map_type_info);
```
Comment (xiaokang) [inline be/src/olap/types.cpp:262]: consider ARRAY

Comment (amorynan) [reply be/src/olap/types.cpp:262]: ARRAY did not has this problem , get_array_type_info(child_column->type(), iterations) which return a with  
```static const ArrayTypeInfo*``` which is static .

**Comment by amorynan** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.07 seconds
 stream load tsv:          430 seconds loaded 74807831229 Bytes, about 165 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          60 seconds loaded 1101869774 Bytes, about 17 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230425062256_clickbench_pr_134553.html

**Comment by amorynan** [comment]
run p0

**Comment by amorynan** [comment]
run p0

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run p1

**Review by yiguolei — APPROVED** [review]
lgtm

**Pull request merged**